File: shells/ios-template/WefterBridge/GeneratedRegistry.swift

import Foundation
import UIKit

final class DeviceInfoPluginDispatch: NativeModule {
    private let plugin: DeviceInfoPlugin
    init(plugin: DeviceInfoPlugin) { self.plugin = plugin }

    func invoke(method: String, payload: [String: Any], callback: @escaping (Result<Any, Error>) -> Void) throws {
        switch method {
        case "getInfo":
            try plugin.getInfo(payload: payload, callback: callback)
        default:
            callback(.failure(WefterError(code: "UNKNOWN_METHOD", message: "No such method: \(method)")))
        }
    }
}

final class PingTestPluginDispatch: NativeModule {
    private let plugin: PingTestPlugin
    init(plugin: PingTestPlugin) { self.plugin = plugin }

    func invoke(method: String, payload: [String: Any], callback: @escaping (Result<Any, Error>) -> Void) throws {
        switch method {
        case "ping":
            try plugin.ping(payload: payload, callback: callback)
        default:
            callback(.failure(WefterError(code: "UNKNOWN_METHOD", message: "No such method: \(method)")))
        }
    }
}

enum GeneratedRegistry {
    static func registerAll(dispatcher: BridgeDispatcher, viewController: UIViewController) {
        let deviceInfoPlugin = DeviceInfoPlugin(dispatcher: dispatcher, viewController: viewController)
        dispatcher.register("device-info", module: DeviceInfoPluginDispatch(plugin: deviceInfoPlugin))
        let pingTestPlugin = PingTestPlugin(dispatcher: dispatcher, viewController: viewController)
        dispatcher.register("ping-test", module: PingTestPluginDispatch(plugin: pingTestPlugin))
    }
}

File: shells/android-template/app/src/main/java/dev/wefter/bridge/GeneratedRegistry.kt

package dev.wefter.bridge

import android.content.Context
import org.json.JSONObject

class DeviceInfoPluginDispatch(private val plugin: DeviceInfoPlugin) : NativeModule {
    override fun invoke(method: String, payload: JSONObject, callback: (Result<Any>) -> Unit) {
        when (method) {
            "getInfo" -> plugin.getInfo(payload, callback)
            else -> callback(Result.failure(WefterError("UNKNOWN_METHOD", "No such method: $method")))
        }
    }
}

class PingTestPluginDispatch(private val plugin: PingTestPlugin) : NativeModule {
    override fun invoke(method: String, payload: JSONObject, callback: (Result<Any>) -> Unit) {
        when (method) {
            "ping" -> plugin.ping(payload, callback)
            else -> callback(Result.failure(WefterError("UNKNOWN_METHOD", "No such method: $method")))
        }
    }
}

object GeneratedRegistry {
    fun registerAll(context: Context, dispatcher: BridgeDispatcher) {
        val deviceInfoPlugin = DeviceInfoPlugin(context, dispatcher)
        dispatcher.register("device-info", DeviceInfoPluginDispatch(deviceInfoPlugin))
        val pingTestPlugin = PingTestPlugin(context, dispatcher)
        dispatcher.register("ping-test", PingTestPluginDispatch(pingTestPlugin))
    }
}
